src/parse.coffee

```diff
@@ -33,8 +33,7 @@ parse = (source, callback) ->
       err = new Error(err.message)
 
     if apiElements
-      apiElements = apiElements.toRefract()
-      apiElements.content = apiElements.content.concat(annotations)
+      apiElements.content.unshift(warning) if warning
```

Comment:
Yes there are tests (https://github.com/refractproject/minim/commit/c18eaa48536baad7481e8debcc1c8fb95ced5ce6). It will be important to switch to the convience functions for appending/removing elements from arrays especially when we move to having a `parent` property in the future to keep the states intact.

---

I suspect this problem is happening because you have a separate minim namespace than the one used by fury. The annotation instance created is from a dredd specific minim instance (`minim = require('minim').namespace().use(minimParseResult)`).

This means, when the `unshift` function checks if the given argument is an element (inherits from BaseElement) the check is coming back false because it isn't actually an instance of "BaseElement" but an instance of another runtime class called "BaseElement". Since Minim determines that the given argument is not an element, it will then attempt to convert the "JSON Object" into an element by deserialising the object. This is why you can see the new `object` element which has a `member` called `annotation` etc. It is a refracted element representation of the annotation (double refraction).

This is an artefact of how the JavaScript classes are constructed (because JavaScript doesn't have classes[1]), when minim is imported from the script eventually `createClass` is called to create a new runtime meta class (https://github.com/refractproject/minim/blob/master/lib/primitives/base-element.js#L8).

As I see it there are two solutions:

1) Fury could expose the `minim` namespace so that consumers can grab hold of it and use it.
2) We move minim to ES6 and away from uptown so that the runtime classes would be equivalent. I am not sure the reasons why uptown was picked, and perhaps there is a reason to keep using uptown.

[1]: In older versions supported by browsers*